app/gui/src/controller/searcher/component.rs

```diff
@@ -175,7 +164,10 @@ impl List {
             component.update_matching_info(pattern)
         }
         for group in self.all_groups() {
-            group.update_sorting(pattern);
+            group.update_sorting_and_visibility(pattern);
+        }
+        for group in self.favorites.iter() {
+            group.update_visibility();
         }
```

Comment:
The `all_groups` should return all groups - including favorites. If we don't want to, the method should be renamed.

Reply:
Actually, it appears that the name `all_groups` was not correct even before: it did not include the groups from the `top_modules` field either.

Given this, and that `all_groups` was only used in one place, I decided to inline its body. I believe this makes the code more clear and explicit as to what it is actually doing.

Reply:
> Actually, it appears that the name `all_groups` was not correct even before: it did not include the groups from the `top_modules` field either.

That's not true. There was no need of iterating over top_modules, as they repeated the groups contained in the map. Only flattened submodules have no representation in that map (perhaps it should be commented somewhere).